cypress/support/commands.js

```diff
@@ -48,4 +48,17 @@ Cypress.on('uncaught:exception', (err, runnable) => {
            });
      }
     });
-  });
\ No newline at end of file
+  });
+  Cypress.Commands.add("typeAndEnter", (selector, text) => {
+    cy.get(selector).should('be.visible').type(text);
+    cy.get(selector).type('{enter}');
+    
+  });
+  Cypress.Commands.add('checkSearchResults', (expectedResultCount, searchTerm) => {
+    cy.get('span.gf-summary b').should('have.text', expectedResultCount).then((resultCount) => {
+      if (resultCount !== expectedResultCount) {
+        cy.log(`Expected Result: ${expectedResultCount}, Actual result: ${resultCount}`);
+      }
```

Comment:
Instead of logging error reason with if condition, you can give this as message in assertion as third parameter